### PR TITLE
Itkdev - CMN-630 - add listview to url

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://jira.itkdev.dk/browse/CMN-630

#### Description

add listview to url (for statistics)
add RecommenderView type
save RecommenderView in url (for statistics)
remove dash from creation of modal-url (due date had two dashes, that annoyed me)
removed unused faustIdModalQueryParam

#### Screenshot of the result

n/a

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

n/a

### CODE REVIEW

**Review by yepzdk — APPROVED** [review]

File: src/apps/something-similar/SomethingSimilarList.tsx [hunk 1]
```diff
@@ -25,15 +26,23 @@ const SomethingSimilarList: FC<SomethingSimilarListProps & MaterialProps> = ({
...
+  const [recommendView, setRecommendView] =
+    useState<RecommenderView>("similar");
   const { data: somethingSimilarData } = useRecommendFromFaustQuery({
     faust: id,
     limit: somethingSimilarLimit as number
   });
 
+  const setViewHandler = useCallback(
+    (inputView: RecommenderView) => {
+      window.history.pushState("", "", `?recommender-view=${inputView}`);
```
Comment (spaceo) [inline src/apps/something-similar/SomethingSimilarList.tsx:40]: Are you sure yow want to use pushState and thereby add to the browser history. Or do you want to rewrite the history? In that case you should use `setQueryParametersInUrl`.
But if you want to add to the history I think you should do following:

* Create a function for it in the [url helper file ](https://github.com/reload/dpl-react/blob/604f25f4a85f9ed6b9e6e9e542291aa7a9f76f18/src/core/utils/helpers/url.ts).
* I think it is a bad practise to hardcode query params like that. What if another url param has already been set by some other functionality? It is move safe to use the [searchParam API](https://github.com/reload/dpl-react/blob/604f25f4a85f9ed6b9e6e9e542291aa7a9f76f18/src/core/utils/helpers/url.ts#L31)

File: src/core/utils/helpers/general.ts [hunk 2]
```diff
@@ -415,7 +415,7 @@ export const patronAgeValid = (cpr: string, minAge: number) => {
 };
 
 export const constructModalId = (prefix: string, fragments: string[]) =>
-  `${prefix ? `${prefix}-` : ""}${fragments.join("-")}`;
+  `${prefix ? `${prefix}` : ""}${fragments.join("-")}`;
```
Comment (spaceo) [inline src/core/utils/helpers/general.ts:418]: I don't get this change. Why shouldn't the prefix be followed by a separator?
Actually I have come up with a more nice way to define this function:
```typescript
export const constructModalId = (prefix: string, fragments: string[]) =>
  `${(prefix ? [prefix, ...fragments] : fragments).join('-')}`;

File: src/components/reservation/ReservationModal.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ const ReservationModal = ({
         "reservationModalScreenReaderModalDescriptionText"
       )}
       closeModalAriaLabelText={t("reservationModalCloseModalAriaLabelText")}
-      dataCy={dataCy || "reservation-modal"}
+      dataCy={dataCy || "-reservation-modal-"}
```
Comment (spaceo) [inline src/components/reservation/ReservationModal.tsx:34]: Why is this necessary? It does not look right to me.

**Review by spaceo — CHANGES_REQUESTED** [review]

File: src/components/reservation/ReservationModal.tsx (hunk 3 above)
Comment (sinejespersen) [reply src/components/reservation/ReservationModal.tsx:34]: No, it isnt necessary, i can revert the code with the dashes :)

File: src/apps/something-similar/SomethingSimilarList.tsx (hunk 1 above)
Comment (sinejespersen) [reply src/apps/something-similar/SomethingSimilarList.tsx:40]: I'll try setQueryParametersInUrl :)

**Review by spaceo — APPROVED** [review]
LGTM 👍
